sources/packages/api/src/services/student/student.service.ts

```diff
@@ -174,44 +186,102 @@ export class StudentService extends RecordDataModelService<Student> {
    * @returns Students pending SIN validation.
    */
   async getStudentsPendingSinValidation(): Promise<Student[]> {
-    return this.repo.find({ validSIN: null });
+    return this.repo
+      .createQueryBuilder("student")
+      .select([
+        "student.id",
+        "student.sin",
+        "student.birthDate",
+        "user.firstName",
+        "user.lastName",
+        "user.id",
+        "sinValidations.id",
+      ])
+      .innerJoin("student.user", "user")
+      .innerJoin("student.sinValidation", "sinValidations")
+      .where("sinValidations.isValidSIN is null")
+      .andWhere("sinValidations.dateSent is null")
+      .andWhere("sinValidations.dateReceived is null")
+      .getMany();
   }
 
   /**
    * Gets a student's valid sin status
    * @returns Students SIN validation status.
    * @param userId information needed to select the user.
    */
-  async getStudentSinStatus(userId: number): Promise<Student> {
-    return await this.repo
+  async getStudentSinStatus(userId: number): Promise<boolean> {
+    const student = await this.repo
       .createQueryBuilder("student")
       .innerJoin("student.user", "user")
+      .innerJoin("student.sinValidation", "sinValidation")
       .where("user.id= :userId ", { userId })
-      .select("student.validSIN")
+      .select("sinValidation.isValidSIN")
       .getOne();
+    return student?.sinValidation.isValidSIN;
   }
 
   /**
    * Update the SIN validation information on a student that
    * is marked with a pending validation.
-   * If the SIN validation status is already set, it needs to be
-   * set as NULL before it is processed again.
-   * @param sin
-   * @param validSIN
-   * @returns pending sin validation
+   * @param craRecord
+   * @param fileReceived
    */
   async updatePendingSinValidation(
-    sin: string,
-    validSIN: boolean,
+    verificationId: number,
+    craRecord: CRAResponseStatusRecord,
+    fileReceived: string,
   ): Promise<void> {
-    // Only allow updates on a student that has pending SIN validation.
-    const studentToUpdate = await this.repo.findOne({
-      sin,
-      validSIN: null,
-    });
-    if (studentToUpdate) {
-      studentToUpdate.validSIN = validSIN;
-      this.repo.save(studentToUpdate);
+    const student = await this.repo
```

Comment:
I do not see a reason for this query being performed on students. We need only to update the sin_validations table based on the sin_validations record id. Am I missing something?